joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/model/GoModTest.scala

```diff
@@ -0,0 +1,129 @@
+package io.joern.go2cpg.model;
+
+import io.joern.gosrc2cpg.Config
+import io.joern.gosrc2cpg.model.{GoMod, GoModDependency, GoModModule}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import java.nio.file.Paths
+
+class GoModTest extends AnyWordSpec with Matchers with BeforeAndAfterAll {
+  "go project without .mod file" in {
+    val namespace = GoMod.getNameSpace(Paths.get("").toAbsolutePath.toString, "main")
+    namespace shouldBe "main"
+  }
+  "invalid compilation file unit with main pkg" in {
+    val config = Config()
+    config.inputPath = Paths.get("").toAbsolutePath.toString
+    GoMod.config = config
+    GoMod.meta = GoMod(
+      fileFullPath = Paths.get(config.inputPath, "go.mod").toAbsolutePath.toString,
+      module = GoModModule("joern.io/trial"),
+      dependencies = List[GoModDependency]()
+    )
+    var namespace =
+      GoMod.getNameSpace(null, "main")
+    namespace shouldBe "main"
+    namespace = GoMod.getNameSpace("", "main")
+    namespace shouldBe "main"
+
+  }
+  "with .mod file and main pkg 1 use case" in {
+    val config = Config()
+    config.inputPath = Paths.get("").toAbsolutePath.toString
+    GoMod.config = config
+    GoMod.meta = GoMod(
+      fileFullPath = Paths.get(config.inputPath, "go.mod").toAbsolutePath.toString,
+      module = GoModModule("joern.io/trial"),
+      dependencies = List[GoModDependency]()
+    )
+    val namespace =
+      GoMod.getNameSpace(Paths.get(config.inputPath, "first", "second", "test.go").toAbsolutePath.toString, "main")
+    namespace shouldBe "first/second/main"
+  }
+
+  "with .mod file and main pkg 2 use case" in {
+    val config = Config()
+    config.inputPath = Paths.get("").toAbsolutePath.toString + "/"
+    GoMod.config = config
+    GoMod.meta = GoMod(
+      fileFullPath = Paths.get(config.inputPath, "go.mod").toAbsolutePath.toString,
+      module = GoModModule("joern.io/trial"),
+      dependencies = List[GoModDependency]()
+    )
+    val namespace =
+      GoMod.getNameSpace(Paths.get(config.inputPath, "first", "second", "test.go").toAbsolutePath.toString, "main")
```

Comment:
With `better.io.Files` you can also use
```scala
import better.files.File
import scala.language.postfixOps

GoMod.getNameSpace(File(config.input) / "first" / "second" / "test.go" pathAsString, "main")
```

Reply:
Something interesting to learn, I appreciate the suggestions :)

Reply:
done